plugins/taper/run.go

```diff
@@ -123,6 +134,10 @@ func (a *Agent) startReadOutputItems(ctx context.Context, outputItems chan *api.
 					log.Errorf("Failed to create telemetry: %v", err)
 					return
 				}
+				if a.traceSamplingManager != nil && !a.traceSamplingManager.ShouldTrace(telemetry.Request.Host) {
+					log.Infof("Ignoring host: %v", telemetry.Request.Host)
+					return
+				}
```

Comment:
shouldn't it be part of `shouldIgnoreTelemetry`?

Reply:
Dont think so. shouldIgnoreTelemetry is a different logic of ignoring, it is more like filtering. 
Also, createTelemetry is extracting the host, I didnt want to extract it twice